### PR TITLE
Allow use of "arguments" on ambient context

### LINKED ISSUE #15390: Allow `arguments` name on parameter in ambient context
**TypeScript Version:**  2.2.1

**Code**

```ts
// A *self-contained* demonstration of the problem follows...
declare class HTMLCanvasElement {
    toBlob(callback: (result: Blob | null) => void, type?: string, ...arguments: any[]): void;
}
```

**Expected behavior:** The name `arguments` on ambient class declarations does not really do anything, it should just be allowed.

**Actual behavior:** This is blocked.

### PR DESCRIPTION
**Pull request by SaschaNaz**
Fixes #15390

### CODE REVIEW

File: src/compiler/binder.ts [hunk 1]
```diff
@@ -2503,8 +2503,27 @@ namespace ts {
...
+            }
+
+            if (node.parent.parent && (
+                node.parent.parent.kind === SyntaxKind.InterfaceDeclaration ||
+                node.parent.parent.kind === SyntaxKind.TypeLiteral
+            )) {
+                return false;
+            }
+
+            if (isDeclarationFile(file) || isInAmbientContext(node)) {
```
Comment (DanielRosenwasser) [inline src/compiler/binder.ts]: Does `isInAmbientContext` cover the above cases?

Comment (SaschaNaz) [inline src/compiler/binder.ts]: I hope so but it only checks whether parents node has `declare` modifiers. I first tried modifying it but it broke other tests.

File: src/compiler/binder.ts [hunk 2]
```diff
@@ -2503,8 +2503,27 @@ namespace ts {
             }
         }
 
+        function needParameterStrictModeCheck(node: Node) {
+            if (node.parent.kind === SyntaxKind.ConstructorType || node.parent.kind === SyntaxKind.FunctionType) {
```
Comment (mhegazy) [inline src/compiler/binder.ts]: i do not think this is important. i would say if this is your code, and this is a .ts file and your writing an interface with `arguments`, change it.

the issue is more of a problem with .d.ts files, so i would just use `isInAmbientContext`.

File: src/compiler/binder.ts (hunk 1 above)
Comment (mhegazy) [inline src/compiler/binder.ts]: isInAmbientContext actually checks the parent file at the end.

Also do not use `file` directly, you do not know what file you are checking, and what it cases the checker to jump into. no guarantees that the checker proceeds to check all nodes in a file before jumping to another.

File: src/compiler/binder.ts (hunk 2 above)
Comment (SaschaNaz) [inline src/compiler/binder.ts]: I think why not, but I agree this is not strictly necessary. Done!

**Comment by SaschaNaz** [comment]
Requested change is done, is there any more requirement?

**Pull request merged**